cloudbuild.yaml

```diff
@@ -3,16 +3,49 @@ steps:
     args:
       - build
       - '--build-arg'
-      - 'DOCKER_FLOW_VERSION=${_VERSION}'
+      - DOCKER_FLOW_VERSION=${_VERSION}
       - '-t'
-      - '${_IMAGE_NAME}'
+      - ${_IMAGE_NAME}:${_IMAGE_VERSION}
       - .
-timeout: 1500s
+    id: Build
+  - name: gcr.io/cloud-builders/docker
+    args:
+      - push
+      - ${_IMAGE_NAME}:${_IMAGE_VERSION}
+    id: Push
+  - name: bash
+    args:
+      - '-c'
+      - echo "$$DEV_CONFIG" > /workspace/dev-config.yml && echo "$$INGRESS" > /workspace/infrastructure/k8s/templates/ingress.yaml
+    id: Get secrets
+    secretEnv:
+      - DEV_CONFIG
+      - INGRESS
+  - name: node:${_NODE_VERSION}
+    args:
+      - '-c'
+      - curl https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3 | bash && curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl" && install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl && echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] http://packages.cloud.google.com/apt cloud-sdk main" | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list && curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key --keyring /usr/share/keyrings/cloud.google.gpg add - && apt-get update -y && apt-get install google-cloud-cli google-cloud-sdk-gke-gcloud-auth-plugin -y && gcloud container clusters get-credentials ${PROJECT_ID} --region us-east1 --project ${PROJECT_ID} && yarn && yarn dev:db:upgrade && yarn dev:apply
+    id: Deploy
+    entrypoint: bash
+  - name: gcr.io/cloud-builders/gcloud
+    args:
+      - '-c'
+      - gcloud container images list-tags ${_IMAGE_NAME} --filter='-tags:*' --format='get(digest)' --limit=unlimited | xargs -I {sha} gcloud container images delete "${_IMAGE_NAME}@{sha}"
```

Comment:
hmm. how does this work? Is it excluding currently built one? Or registry removes some tags from older images, so only those are being listed?

Reply:
Registry only ever has one `latest` tag and this is filtering on images with _no_ tags (so, not `latest`). It'll get that list and delete all of them.